From 9c5e8d1f018e5118108ef9273412142efc2ceb4d Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Sat, 24 Feb 2018 19:35:07 +0000
Subject: [PATCH 3/3] sepolicy: Add rules for LiveDisplay HIDL HAL

 * The HIDL HAL replaces the JNI implementation, so remove the old rules.

Change-Id: I9b83480633041122c5093d35f31d2ecc1f61149a
---
 common/vendor/file_contexts              |  3 +++
 common/vendor/hal_livedisplay_default.te | 12 ++++++++++++
 common/vendor/hwservice.te               |  2 ++
 common/vendor/hwservice_contexts         |  2 ++
 common/vendor/system_server.te           |  3 +++
 qcom/common/hal_livedisplay_default.te   | 11 +++++++++++
 qcom/common/system_server.te             |  3 ---
 7 files changed, 33 insertions(+), 3 deletions(-)
 create mode 100644 common/vendor/hal_livedisplay_default.te
 create mode 100644 qcom/common/hal_livedisplay_default.te
 delete mode 100644 qcom/common/system_server.te

diff --git a/common/vendor/file_contexts b/common/vendor/file_contexts
index 3965b86..db48178 100644
--- a/common/vendor/file_contexts
+++ b/common/vendor/file_contexts
@@ -4,5 +4,8 @@
 # Light HAL
 /(vendor|system/vendor)/bin/hw/android\.hardware\.light@2\.0-service\.aw2013 u:object_r:hal_light_default_exec:s0
 
+# LiveDisplay HAL
+/(vendor|system/vendor)/bin/hw/vendor\.lineage\.livedisplay@1\.0-service-sdm u:object_r:hal_livedisplay_default_exec:s0
+
 # Vibrator HAL
 /(vendor|system/vendor)/bin/hw/android\.hardware\.vibrator@1\.0-service\.lineage u:object_r:hal_vibrator_default_exec:s0
diff --git a/common/vendor/hal_livedisplay_default.te b/common/vendor/hal_livedisplay_default.te
new file mode 100644
index 0000000..874d446
--- /dev/null
+++ b/common/vendor/hal_livedisplay_default.te
@@ -0,0 +1,12 @@
+type hal_livedisplay_default, domain;
+hwbinder_use(hal_livedisplay_default)
+vndbinder_use(hal_livedisplay_default)
+
+type hal_livedisplay_default_exec, exec_type, vendor_file_type, file_type;
+init_daemon_domain(hal_livedisplay_default)
+
+# Check if hwservicemanager is ready
+get_prop(hal_livedisplay_default, hwservicemanager_prop)
+
+# Add vendor.lineage.livedisplay::IColor service to hwservicemanager
+add_hwservice(hal_livedisplay_default, hal_lineage_livedisplay_hwservice)
diff --git a/common/vendor/hwservice.te b/common/vendor/hwservice.te
index 2ca8189..8f1a791 100644
--- a/common/vendor/hwservice.te
+++ b/common/vendor/hwservice.te
@@ -1 +1,3 @@
+type hal_lineage_livedisplay_hwservice, hwservice_manager_type;
+
 type hal_lineage_power_hwservice, hwservice_manager_type;
diff --git a/common/vendor/hwservice_contexts b/common/vendor/hwservice_contexts
index 3ae7fd7..193b712 100644
--- a/common/vendor/hwservice_contexts
+++ b/common/vendor/hwservice_contexts
@@ -1 +1,3 @@
+vendor.lineage.livedisplay::IColor       u:object_r:hal_lineage_livedisplay_hwservice:s0
+
 vendor.lineage.power::ILineagePower      u:object_r:hal_lineage_power_hwservice:s0
diff --git a/common/vendor/system_server.te b/common/vendor/system_server.te
index 4a889fc..942a293 100644
--- a/common/vendor/system_server.te
+++ b/common/vendor/system_server.te
@@ -1 +1,4 @@
+allow system_server hal_lineage_livedisplay_hwservice:hwservice_manager find;
+binder_call(system_server, hal_livedisplay_default)
+
 allow system_server hal_lineage_power_hwservice:hwservice_manager find;
diff --git a/qcom/common/hal_livedisplay_default.te b/qcom/common/hal_livedisplay_default.te
new file mode 100644
index 0000000..3c2dacc
--- /dev/null
+++ b/qcom/common/hal_livedisplay_default.te
@@ -0,0 +1,11 @@
+# Do not use add_service() as hal_graphics_composer_default may be the provider as well
+allow hal_livedisplay_default qdisplay_service:service_manager find;
+
+binder_call(hal_livedisplay_default, hal_graphics_composer_default)
+
+# Allow LiveDisplay to store files under /data/vendor/display and access them
+allow hal_livedisplay_default display_misc_file:dir rw_dir_perms;
+allow hal_livedisplay_default display_misc_file:file create_file_perms;
+
+# Allow LiveDisplay to access pps socket
+unix_socket_connect(hal_livedisplay_default, pps, mm-pp-daemon)
diff --git a/qcom/common/system_server.te b/qcom/common/system_server.te
deleted file mode 100644
index 7857eaa..0000000
--- a/qcom/common/system_server.te
+++ /dev/null
@@ -1,3 +0,0 @@
-# Livedisplay
-allow system_server display_misc_file:dir rw_dir_perms;
-allow system_server display_misc_file:file create_file_perms;
-- 
2.17.0

